src/riak_object.erl

```diff
@@ -1104,7 +1090,9 @@ decode_msgpack(ValBin) ->
 decode_maybe_binary(<<1, Bin/binary>>) ->
     Bin;
 decode_maybe_binary(<<0, Bin/binary>>) ->
-    binary_to_term(Bin).
+    binary_to_term(Bin);
+decode_maybe_binary(Bin) ->
+    decode_msgpack(Bin).
 
```

Comment:
Need to ensure that we review the stability of this change under this JIRA

https://bashoeng.atlassian.net/browse/RTS-333